src/main/java/com/servicewizard/model/ServiceMethod.java

```diff
@@ -54,51 +59,81 @@ public void setDescription(String description) {
 	/**
 	 * The verb describing the service method's HTTP request method
 	 */
-	public HttpVerb getVerb() {
+	public String getVerb() {
 		return verb;
 	}
 
 	/**
-	 * Sets the HTTP verb for this method
-	 */
-	public void setVerb(HttpVerb verb) {
-		this.verb = verb;
-	}
-
-	/**
 	 * The URL for this method, relative to the host it is served on
 	 */
-	public String getRelativePath() {
-		return relativePath;
+	public String getPath() {
+		return path;
 	}
 
 	/**
 	 * Sets the URL for this method, relative to the host it is served on
 	 */
-	public void setRelativePath(String relativePath) {
-		this.relativePath = relativePath;
+	public void setPath(String path) {
+		this.path = path;
+	}
+
+	public boolean isRequiresAuthentication() {
+		return requiresAuthentication;
+	}
+
+	public void setRequiresAuthentication(boolean requiresAuthentication) {
+		this.requiresAuthentication = requiresAuthentication;
+	}
+
+	public String getPermissionRequired() {
+		return permissionRequired;
+	}
+
+	public void setPermissionRequired(String permissionRequired) {
+		this.permissionRequired = permissionRequired;
 	}
 
 	/**
 	 * The HTTP query parameters accepted by this method
 	 */
-	public List<String> getQueryParameters() {
+	public List<ServiceMethodParameter> getQueryParameters() {
 		return queryParameters;
 	}
 
 	/**
 	 * Add an HTTP parameter this service method will accept
 	 */
-	public void addQueryParameter(String param) {
+	public void addQueryParameter(ServiceMethodParameter param) {
 		queryParameters.add(param);
 	}
 
+	public List<String> getQueryParameterNames() {
+		return queryParameters.stream()
+				.map(p -> p.getName())
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * The path parameters required by this method
+	 */
+	public List<ServiceMethodParameter> getPathParameters() {
+		return pathParameters;
+	}
+
+	/**
+	 * Add a path parameter this service method requires.
+	 */
+	public void addPathParameter(ServiceMethodParameter param) {
+		pathParameters.add(param);
+	}
+
 	/**
 	 * Whether the service method requires a request body.
 	 *
 	 * POST methods will typically have a request body, while GET methods typically will not.
 	 */
-	public boolean hasRequestBody() {
+	// name sucks - hooray bean patterns!
```

Comment:
I appreciate the humor but I don't want this comment here :) If we feel it's important to justify the name to future developers, we could include a note in a project style guide that says to follow the bean pattern.

Also I think is- is an accepted bean prefix for boolean values.